official/vision/beta/projects/yolo/modeling/backbones/Darknet.py

```diff
@@ -0,0 +1,356 @@
+      input_shape=tf.keras.layers.InputSpec(shape=[None, None, None, 3]),
+      min_size=None,
+      max_size=5,
+      activation=None,
+      use_sync_bn=False,
+      norm_momentum=0.99,
+      norm_epsilon=0.001,
+      kernel_initializer='glorot_uniform',
+      kernel_regularizer=None,
+      bias_regularizer=None,
+      config=None,
+      **kwargs):
+
+    layer_specs, splits = Darknet.get_model_config(model_id)
+
+    self._model_name = model_id
+    self._splits = splits
+    self._input_shape = input_shape
+    self._registry = layer_registry()
+
+    # default layer look up
+    self._min_size = min_size
+    self._max_size = max_size
+    self._output_specs = None
+
+    self._kernel_initializer = kernel_initializer
+    self._bias_regularizer = bias_regularizer
+    self._norm_momentum = norm_momentum
+    self._norm_epislon = norm_epsilon
+    self._use_sync_bn = use_sync_bn
+    self._activation = activation
+    self._weight_decay = kernel_regularizer
+
+    self._default_dict = {
+        "kernel_initializer": self._kernel_initializer,
+        "weight_decay": self._weight_decay,
+        "bias_regularizer": self._bias_regularizer,
+        "norm_momentum": self._norm_momentum,
+        "norm_epsilon": self._norm_epislon,
+        "use_sync_bn": self._use_sync_bn,
+        "activation": self._activation,
+        "name": None
+    }
+
+    inputs = ks.layers.Input(shape=self._input_shape.shape[1:])
+    output = self._build_struct(layer_specs, inputs)
+    super().__init__(inputs=inputs, outputs=output, name=self._model_name)
+    return
+
+  @property
+  def input_specs(self):
+    return self._input_shape
+
+  @property
+  def output_specs(self):
+    return self._output_specs
+
+  @property
+  def splits(self):
+    return self._splits
+
+  def _build_struct(self, net, inputs):
+    endpoints = collections.OrderedDict()
+    stack_outputs = [inputs]
+    for i, config in enumerate(net):
+      if config.stack == None:
+        x = self._build_block(stack_outputs[config.route],
+                              config,
+                              name=f"{config.layer}_{i}")
+        stack_outputs.append(x)
+      elif config.stack == "residual":
+        x = self._residual_stack(stack_outputs[config.route],
+                                 config,
+                                 name=f"{config.layer}_{i}")
+        stack_outputs.append(x)
+      elif config.stack == "csp":
+        x = self._csp_stack(stack_outputs[config.route],
+                            config,
+                            name=f"{config.layer}_{i}")
+        stack_outputs.append(x)
+      elif config.stack == "csp_tiny":
+        x_pass, x = self._tiny_stack(stack_outputs[config.route],
+                                     config,
+                                     name=f"{config.layer}_{i}")
+        stack_outputs.append(x_pass)
+      if (config.is_output and
+          self._min_size == None):
+        endpoints[config.output_name] = x
```

Comment:
We need output_name to be str, so can you cast it to str everywhere?

Reply:
addressed